index.js

```diff
@@ -164,121 +166,101 @@ FeatureService.prototype.getObjectIdField = function (info) {
  */
 FeatureService.prototype.layerIds = function (callback) {
   this.request(this.url + '/' + this.layer + '/query?where=1=1&returnIdsOnly=true&f=json', function (err, json) {
+    if (!json.objectIds) return callback(new Error('Request for object ids failed'))
+    // TODO: is this really necessary
+    json.objectIds.sort(function (a, b) { return a - b })
     callback(err, json.objectIds)
```

Comment:
should catch and return `err` first then if all's well end with `callback(null, json.objectIds)`